vars/prepareAndRunAnsiblePlaybook.groovy

```diff
@@ -0,0 +1,39 @@
+def call(Map params) {
+  def playbook = params.playbook
+  def prepareConfig = params.config
+
+  pipeline {
+    agent none
+    stages {
+      stage('Prepare environment') {
+        agent { label 'ansible' }
+        environment {
+          VC = credentials('vcenter')
+        }
+        steps {
+          dir('ansible') {
+            createCommonVars env.SHARED_DRIVE_IP, env.JENKINS_MASTER_IP
+            createAnsibleConfig env.ANSIBLE_VAULT_KEY_FILE
+            sh 'cp inventory.sample inventory'
+            sh 'ansible-galaxy install -r requirements.yml -f'
+            script {
```

Comment:
Does it have to be in a script block?

Reply:
Unfortunately, it has to. Jenkins complained that it cannot run function passed as an argument outside the script block.

Reply:
:+1: